src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
```

Comment:
It's better to use some() method for finding person with suitable mother.
Don't use single letter 'p' or 'p1' for variables naming. It could be 'person' and 'personToCompare' accordingly for example. You also have another option to define people list in nested some() (in your code find()) method. You can write  
`people.filter((person, index, array) => person.sex === 'f'
   ? array.find((personToCompare) => personToCompare.mother === person.mother))
`